s3/src/main/java/com/yahoo/ycsb/db/S3Client.java

```diff
@@ -0,0 +1,305 @@
+	* values HashMap will be written into the file with the specified file
+	* key, overwriting any existing values with the same field name.
+	* 
+	* @param bucket
+	*            The name of the bucket
+	* @param key
+	*            The file key of the file to write.
+	* @param values
+	*            A HashMap of field/value pairs to update in the record
+	* @return Zero on success, a non-zero error code on error. See this class's
+	*         description for a discussion of error codes.
+	*/
+	@Override
+	public int update(String bucket, String key,HashMap<String, ByteIterator> values) {
+		return writeToStorage(bucket,key,values,1);
+	}
+	/**
+	* Perform a range scan for a set of files in the bucket. Each
+	* field/value pair from the result will be stored in a HashMap.
+	* 
+	* @param bucket
+	*            The name of the bucket
+	* @param startkey
+	*            The file key of the first file to read.
+	* @param recordcount
+	*            The number of files to read
+	* @param fields
+	*            The list of fields to read, or null for all of them
+	* @param result
+	*            A Vector of HashMaps, where each HashMap is a set field/value
+	*            pairs for one file
+	* @return Zero on success, a non-zero error code on error. See the
+	*         {@link DB} class's description for a discussion of error codes.
+	*/
+	@Override
+	public int scan(String bucket, String startkey, int recordcount, Set<String> fields, Vector<HashMap<String, ByteIterator>> result) {
+		return scanFromStorage(bucket,startkey,recordcount,result);
+	}
+
+	protected int writeToStorage(String bucket,String key,HashMap<String, ByteIterator> values, int updateMarker) {
+		int totalSize = 0;
+		int fieldCount = values.size(); //number of fields to concatenate
+		Object keyToSearch = values.keySet().toArray()[0]; // getting the first field in the values
+		byte[] sourceArray = values.get(keyToSearch).toArray(); // getting the content of just one field
+		int sizeArray = sourceArray.length; //size of each array
+		if (updateMarker == 0){
+			totalSize = sizeArray*fieldCount;
+		} else {
+			try {
+				S3Object object = this.s3Client.getObject(new GetObjectRequest(bucket, key));
+				ObjectMetadata objectMetadata = this.s3Client.getObjectMetadata(bucket, key);
+				int sizeOfFile = (int)objectMetadata.getContentLength();
+				fieldCount = sizeOfFile/sizeArray;
+				totalSize = sizeOfFile;
+			} catch (Exception e){
+				e.printStackTrace();
+            			return 1;
+			}
+		}
+		
+		byte[] destinationArray = new byte[totalSize];
+		int offset = 0;
+    		for (int i = 0; i < fieldCount; i++) {
+        		System.arraycopy(sourceArray, 0, destinationArray, offset, sizeArray);
+        		offset += sizeArray;
+    		}
+		InputStream input = new ByteArrayInputStream(destinationArray);
+		ObjectMetadata metadata = new ObjectMetadata();
+		metadata.setContentLength(totalSize);
+		try {
+			PutObjectResult res = this.s3Client.putObject(bucket,key,input,metadata);
```

Comment:
Yes true. I added an additional check. I check if the method getETag() for the object PutObjectResult return a non null string. If not I return 1, if yes this indicated the upload was successfull.